### PR TITLE
[KOGITO-2905] - Image definition changes in the CRDs

### PR DESCRIPTION
**Pull request by Kaitou786**
The aim of the PR is to change how we specify the image in our CRDs, instead of the image structure defined by user, user will have to define image as string with complete and full values.
See: https://issues.redhat.com/browse/KOGITO-2905
Signed-off-by: Tarun Khandelwal <tarkhand@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Review by vaibhavjainwiz — APPROVED** [review]
LGTM

File: pkg/apis/app/v1alpha1/kogitobuild_types.go [hunk 1]
```diff
@@ -116,7 +116,7 @@ type KogitoBuildSpec struct {
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Base Image"
```
Comment (ricardozanini) [inline pkg/apis/app/v1alpha1/kogitobuild_types.go:116]: You might need to change the description as well :)

File: pkg/apis/app/v1alpha1/kogitoservices_types.go [hunk 2]
```diff
@@ -139,7 +139,7 @@ type KogitoServiceSpec struct {
 	// Image definition for the service. Example: Domain: quay.io, Namespace: kiegroup, Name: kogito-service, Tag: latest.
 	// On OpenShift an ImageStream will be created in the current namespace pointing to the given image.
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
```
Comment (ricardozanini) [inline pkg/apis/app/v1alpha1/kogitoservices_types.go:141]: We need to change the description of the attribute as well.

**Review by ricardozanini — COMMENTED** [review]
Just review the CRD's image attribute and we are good to go. :)

File: test/framework/mappers/cr_to_cli.go [hunk 3]
```diff
@@ -149,7 +148,7 @@ func GetBuildCLIFlags(kogitoBuild *v1alpha1.KogitoBuild) []string {
 		cmd = append(cmd, "--native")
 	}
 
-	image = framework.ConvertImageToImageTag(kogitoBuild.Spec.RuntimeImage)
+	image = kogitoBuild.Spec.RuntimeImage
 	if len(image) > 0 {
 		cmd = append(cmd, "--image-runtime", image)
 	}
```
Comment (sutaakar) [inline test/framework/mappers/cr_to_cli.go:154]: @Kaitou786 Can you please remove this section? Seems to be a copy of line 133-136 (probably a copy paste leftover from some previous adjustment).

**Review by sutaakar — APPROVED** [review]
Looks good to me

**Comment by Kaitou786** [comment]
The above suggestions and changes in the examples are done in this [separate commit](https://github.com/kiegroup/kogito-cloud-operator/pull/580/commits/956ff8789d06c0fde1790ddb877fe72dd2bb76b8)

**Review by spolti — APPROVED** [review]

**Comment by spolti** [comment]
@Kaitou786 please take a look on the lint test.

**Review by ricardozanini — COMMENTED** [review]
I believe we can merge once QE have verified it and/or the CI is back.

**Comment by sutaakar** [comment]
QE verified the PR manually

**Pull request merged**